### PR TITLE
Documentation fixes: ECSRun ambiguities in IAM roles in the ECSRun docstring

### PR DESCRIPTION
**Pull request by anna-geller**
## Changes
- added examples in ECSRun to show when custom IAM roles must be provided and when those can be skipped
- clarified that all roles must be passed using a full ARN
- fixed confusing docs about versioning and local agent

Related to #5110

### CODE REVIEW

**Review by anticorrelator — DISMISSED** [review]

**Comment by madkinsz** [comment]
I'll review this as well. I'll redeploy the docs once this is merged so it goes live immediately.

File: docs/core/about_prefect/why-not-airflow.md [hunk 1]
```diff
@@ -292,7 +292,7 @@ In Prefect Cloud, we have elevated versioned workflows to a first-class concept.
 - versioning automatically occurs when you deploy a flow to a Project that already contains a flow of the same name
 - when a flow is versioned, it gets an incremented version number and any prior versions are automatically archived (which turns off automatic scheduling)
 
-Both of these settings can be customized if you have more complicated versioning requirements. For example, you could specify that any flow is a version of any other flow, regardless of name or project. You could override the automatic version promotion to unarchive and enable old versions (for example, for A/B testing). Or you could use versioning to maintain a history of your workflow without polluting your UI.
+Both of these settings can be customized if you have more complicated versioning requirements. For more information, visit the [Versioning documentation](/orchestration/concepts/flows.html#versioning).
```
Comment (tpdorsey) [inline docs/core/about_prefect/why-not-airflow.md:295]:
```suggestion
Both of these settings can be customized if you have more complicated versioning requirements. For more information, see the documentation for [Flow Versioning](/orchestration/concepts/flows.html#versioning).
```

File: src/prefect/run_configs/ecs.py [hunk 2]
```diff
@@ -55,46 +63,173 @@ class ECSRun(RunConfig):
...
+            when registering a task definition for this task. If you provide a custom
+            `task_definition` that already contains the `execution_role_arn`, then you
+            can skip this argument. But if you don't set any custom `task_definition`,
+            and you supply an ECR `image`, then you need to set an `execution_role_arn`
+            explicitly to grant ECR access.
         - run_task_kwargs (dict, optional): Additional keyword arguments to
-            pass to `run_task` when starting this task. See the
-            [ECS.Client.run_task][3] docs for more information.
+            pass to `run_task` when starting this task. It should be used only for
+            runtime-specific arguments such as `cpu`, `memory`, `cluster`, `launchType`,
```
Comment (tpdorsey) [inline src/prefect/run_configs/ecs.py:80]:
```suggestion
            runtime-specific arguments such as `cpu`, `memory`, `cluster`, or `launchType`,
```

File: src/prefect/run_configs/ecs.py [hunk 3]
```diff
@@ -55,46 +63,173 @@ class ECSRun(RunConfig):
...
 
     ```python
-    flow.run_config = ECSRun()
+    flow.run_config = ECSRun(labels=["prod"])
     ```
 
-    Use the default task definition, but override the image and CPU:
+    2) Use a custom task definition uploaded to S3 as a YAML file. This task definition contains
+    a container named "flow", as well as a custom execution role and task role. The flow should
+    be picked up for execution by agent with label "prod" and should be deployed to a cluster
```
Comment (tpdorsey) [inline src/prefect/run_configs/ecs.py:97]:
```suggestion
    be picked up for execution by an agent with label "prod" and should be deployed to a cluster
```

**Review by tpdorsey — DISMISSED** [review]
I'm not sure how the numbered examples are going to look in the published documentation, but we can revisit later if necessary. These are great additions. Thanks.

**Comment by anna-geller** [comment]
> I'm not sure how the numbered examples are going to look in the published documentation, but we can revisit later if necessary. These are great additions. Thanks.

@tpdorsey Good point, I also wasn't 100% sure but looking at e.g. [Databricks examples within a docstring](https://docs.prefect.io/api/latest/tasks/databricks.html#databrickssubmitrun) I hope this should be fine.

**Comment by tpdorsey** [comment]
Previewed locally and I'm okay with these changes

**Review by madkinsz — APPROVED** [review]

**Pull request merged**